File: src/main/java/com/company/store/Item.java
package com.company.store;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import javax.persistence.*;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.util.Objects;

@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
@Table(name="items")
public class Item {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @NotEmpty
    private String name;
    @NotEmpty
    private String description;
    @NotNull
    private Double price;
    private Boolean imported;
    @NotEmpty
    private String category;
    @NotEmpty
    private String itemImageUrl;
    @NotNull
    private Integer quantity;

    public Item() {
    }

    public Item(String name, String description, Double price, Boolean imported, String category, String itemImageUrl, Integer quantity) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.price = price;
        this.imported = imported;
        this.category = category;
        this.itemImageUrl = itemImageUrl;
        this.quantity = quantity;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Double getPrice() {
        return price;
    }

    public void setPrice(Double price) {
        this.price = price;
    }

    public Boolean getImported() {
        return imported;
    }

    public void setImported(Boolean imported) {
        this.imported = imported;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getItemImageUrl() {
        return itemImageUrl;
    }

    public void setItemImageUrl(String itemImageUrl) {
        this.itemImageUrl = itemImageUrl;
    }

    public Integer getQuantity() {
        return quantity;
    }

    public void setQuantity(Integer quantity) {
        this.quantity = quantity;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Item item = (Item) o;
        return Objects.equals(id, item.id) &&
                Objects.equals(name, item.name) &&
                Objects.equals(description, item.description) &&
                Objects.equals(price, item.price) &&
                Objects.equals(imported, item.imported) &&
                Objects.equals(category, item.category) &&
                Objects.equals(itemImageUrl, item.itemImageUrl) &&
                Objects.equals(quantity, item.quantity);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, name, description, price, imported, category, itemImageUrl, quantity);
    }
}

File: src/main/frontend/src/app/item.ts
export class Item {
    id: number;
    name: string;
    description: string;
    price: number;
    imported: boolean;
    category: string;
    itemImageUrl: string;
    quantity: number;

    inStock: number;

    itemSalesTax: number;
    itemImportTax: number;
    subtotal: number;
    totalSalesTax: number;
    totalImportTax: number;
    total: number;

    constructor(name: string, description: string, price: number, imported: boolean, category: string, itemImageUrl: string, quantity: number, inStock?: number) {
        this.name = name;
        this.description = description;
        this.price = price;
        this.imported = imported;
        this.category = category;
        this.itemImageUrl = itemImageUrl;
        this.quantity = quantity;
        this.inStock = inStock;
        this.itemSalesTax = this.itemSalesTax;
        this.itemImportTax = this.itemImportTax;
        this.subtotal = this.subtotal;
        this.totalSalesTax = this.totalSalesTax;
        this.totalImportTax = this.totalImportTax;
        this.total = this.total;
    }
}